### ISSUE TITLE
Is it possible to disable auto-completion of brackets, quotes, etc?

### ISSUE THREAD

**Issue by nonplayer** [RickStrahl/MarkdownMonster#779]
The auto-completion feature is really frustrating when writing long bits of text. I looked through everywhere in the settings but did not see anything that seemed to indicate it could toggle this.

My work process involves a lot of rapid key typing and very little mouse using. The "delete" and "end" keys are not immediate;y under my fingers, and I am trying to avoid having to stop my typing, move to those keys, and then adjust my cursor location. Pretty much anyting that automatically adds a character _after_ the cursor while I'm in the middle of typing (outside of a specific CTRL- command) is something I'd like to be able to turn off.

If this is already an option, I am unable to find it.

For reference, my use case: I am an author, game designer, and video game story developer who works exclusively in Markdown. It's gaining a lot of traction among writers who enjoy minimalist text development systems.

Thanks!

**Comment by RickStrahl** [comment]
Unfortunately not.

The editor has a lot of configuration options but this is not one of them as this is a core feature of the editor component that Markdown Monster uses ([ACE Editor](https://ace.c9.io/)). I've customized the markdown handling some but unfortunately that code is a quagmire of hard to parse regex expressions and I've resigned myself to living with the feature set the editor provides and not customizing beyond that as maintaining changes from the base line is difficult and would impair.

So the short answer is - no I won't fix this. I hear you - but this is not worth the effort as this is the first time this has come up and the default behavior is very likely what 99% expect to have happen in a Markdown editor.

If you want to see this fixed or addressed with an option you can try raising an issue or question on the [Ace Editor repo](https://github.com/ajaxorg/ace/issues). If it gets integrated into the core editor then MM automatically inherits that change next time we update, but traditionally they are not very responsive to syntax specific tweaks.

**Comment by RickStrahl** [comment]
FWIW I've experimented around with some additional ACE editor settings and while I can turn off the quote expansion, it's not a universal setting. There are a few things MM already does like allow disabling list auto-expansion (which inserts the next bullet) which can be disabled optionally.

But I think the solution to this would be if there is some way for ACE to simply not do any expansions. I can't find such a setting though, but I've posted a question on Stackoverflow here:

https://stackoverflow.com/questions/67067904/ace-editor-turn-off-all-auto-completions-and-expansions

**Comment by nonplayer** [comment]
Thanks for looking into this!

**Comment by RickStrahl** [comment]
Helps if the SO post gets upvoted, when someone from ACE gets to it.

**Comment by RickStrahl** [comment]
Ok, so looks like there is a solution that allows disabling behaviors in ACE as was helpfully resolved in the [StackOverFlow Issue](https://stackoverflow.com/questions/67067904/ace-editor-turn-off-all-auto-completions-and-expansions).

With that possible I've added a new editor configuration switch `NoAutoComplete` that basically disables all expansions:

![image](https://user-images.githubusercontent.com/1374013/115084895-34596b80-9ea5-11eb-8066-5b3415404bd7.png)

**Comment by nonplayer** [comment]
Thank you!
